src/App.tsx

```diff
@@ -15,35 +17,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FIELD_LENGTH = SortTypes.length;
+const SORT_FIELD_ALPH = SortTypes.alphabet;
+
+type Goods = (
+  goods: string[],
+  sortField: SortTypes,
+  isReversed: boolean,
+) => string[];
+
+const getPreparedGoods: Goods = function goodsPreparation(
+  goods,
+  sortField,
+  isReversed,
+): string[] {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good, anotherGood) => {
+    switch (sortField) {
+      case SORT_FIELD_LENGTH:
+        return good.length - anotherGood.length;
+      case SORT_FIELD_ALPH:
+        return good.localeCompare(anotherGood);
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortTypes.default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+  const isChanged = () => {
+    return sortField !== SortTypes.default || isReversed;
+  };
```

Comment:
```suggestion
  const isResetButtonVisible = sortField !== SortTypes.default || isReversed;
```
Just store as a variable